### PR TITLE
feat: delete project token

### PR DESCRIPTION
**Pull request by pclata**
Delete a project's token with the Cello DB. Also added are some small list tokens unit tests updates for some cleanup to the unit test helper functions.

### CODE REVIEW

File: service/handlers.go [hunk 1]
```diff
@@ -981,6 +981,37 @@ func (h handler) updateTarget(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+	if err := a.Validate(a.ValidateAuthorizedAdmin(h.env.AdminSecret)); err != nil {
+		h.errorResponse(w, "error unauthorized, invalid authorization header", http.StatusUnauthorized)
+		return
+	}
+
+	ctx := r.Context()
+
+	level.Debug(l).Log("message", "deleting token from database")
```
Comment (pclata) [inline service/handlers.go:1007]: @Acepie @ysmolyar should the DB/CP be checked to see if a project exists?

File: service/handlers.go [hunk 2]
```diff
@@ -981,6 +981,37 @@ func (h handler) updateTarget(w http.ResponseWriter, r *http.Request) {
...
+	if err = h.dbClient.DeleteTokenEntry(ctx, tokenID); err != nil {
```
Comment (Acepie) [inline service/handlers.go:1008]: do we need/want to actually invalidate the token from vault also?

File: service/handlers.go (hunk 1 above)
Comment (Acepie) [reply service/handlers.go:1007]: probably safer to check the project cause rn theres nothing actually stopping me from guessing random tokens right?

Comment (Acepie) [reply service/handlers.go:1007]: which i guess means check the project exists and that the token is for the given project

File: service/handlers.go (hunk 2 above)
Comment (pclata) [reply service/handlers.go:1008]: yes we do 🤦‍♂️

**Review by Acepie — CHANGES_REQUESTED** [review]

File: service/internal/credentials/vault.go [hunk 3]
```diff
@@ -336,6 +340,47 @@ func (v VaultProvider) GetTarget(projectName, targetName string) (types.Target,
 	}, nil
 }
 
+func (v VaultProvider) DeleteProjectToken(tokenID string) error {
+	if v.isAdmin() {
```
[the remark this answers is not in the document]
Comment (pclata) [reply service/internal/credentials/vault.go:344]: Sorry PR wasn't ready yet as I still want to test this locally, but It's the logic that's wrong. Thanks for catching that.

**Review by Acepie — APPROVED** [review]

**Review by Acepie — APPROVED** [review]

**Pull request merged**